src/communication/session.js

```diff
@@ -0,0 +1,40 @@
+
+define(["javascriptHelper"], function (javascriptHelper) {
+    "use strict";
+
+
+    var Session = function (serverUrl, publisher) {
+        this.serverUrl = serverUrl;
+        this.webSocket = new javascriptHelper.WebSocket(serverUrl);
+        this.publisher = publisher;
+    }
+
+
+    Session.prototype.init = function (callback, callbackError) {
+        var thisObject = this;
+
+        this.webSocket.onopen = function (e) {
+            console.log("connection started on " + thisObject.serverUrl + ".");
+            callback(thisObject);
+        }
+
+        this.webSocket.onerror = function (e) {
+            console.error("Error on " + thisObject.serverUrl + ".");
+            callbackError(e);
+        }
+
+        this.webSocket.onclose = function (e) {
+            console.log("connection on " + thisObject.serverUrl + " closed.");
+        }
+    }
+
+
+    Session.prototype.send = function (componentName, stateMachineName, jsonMessage) {
+        var data = this.publisher.getEventToSend(componentName, stateMachineName, jsonMessage);
+        var stringToSend = data.routingKey + " " + data.event.Header.ComponentCode.Fields[0]
```

Comment:
create a private method with good naming to explain the purpose of the following code data.routingKey + " " + data.event.Header.ComponentCode.Fields[0]